torch/distributed/_sharding_spec/api.py

```diff
@@ -37,7 +37,16 @@ def device(self) -> Device:
         """
         return self._device
 
-class ChunkShardingSpec(PlacementSpec):
+
+class ShardingSpec(PlacementSpec):
+    """
+    Base class representing sharding specifications. It is special type of
+    PlacementSpec.
+    """
+    pass
+
+
+class ChunkShardingSpec(ShardingSpec):
```

Comment:
I think we should describe clearly what will happen if the # of elements in the specified sharding dimension cannot be divided by the # of shards. After all, the implementation here is different from the data partitioning behavior of 'DataParallel` for the similar case.